tests/test_chiral_restraints.py

```diff
@@ -0,0 +1,526 @@
+    system = s_top.setup_end_state(ff)
+    U_fn = system.get_U_fn()
+    vols_orig = []
+    frames = simulate_system(U_fn, x0)
+    for f in frames:
+        vol = pyramidal_volume(*f[restr_idxs])
+        vols_orig.append(vol)
+
+    def U_total(x):
+        return U_fn(x) + U_chiral_atom(x, restr_idxs, 1000.0)
+
+    vols_chiral = []
+    frames = simulate_system(U_total, x0)
+    for f in frames:
+        vol = pyramidal_volume(*f[restr_idxs])
+        vols_chiral.append(vol)
+
+    vols_orig = np.array(vols_orig)
+    vols_chiral = np.array(vols_chiral)
+
+    # should be within 5% of 50/50 for the original distribution, the <0 case
+    # is implied
+    assert np.abs(np.mean(vols_orig < 0) - 0.5) < 0.05
+    assert np.abs(np.mean(vols_chiral < 0) - 0.95) < 0.05
+
+    ref_dist = [x for x in vols_orig if x < 0]
+    test_dist = [x for x in vols_chiral if x < 0]
+
+    # should be indistinguishable under KS-test
+    ks, pv = scipy.stats.ks_2samp(ref_dist, test_dist)
+    assert ks < 0.05
+    assert pv > 0.10
+
+    # # useful plotting diagnostics, do not remove.
+    # # compare original distributions
+    # plt.hist(vols_orig, bins=np.linspace(-1, 1, 80), alpha=0.5, label="no_chiral_restr", density=True)
+    # plt.hist(vols_chiral, bins=np.linspace(-1, 1, 80), alpha=0.5, label="with_chiral_restr", density=True)
+    # plt.legend()
+    # plt.title("pyramidal chirality")
+    # plt.xlabel("chiral volume")
+    # plt.ylabel("samples")
+    # plt.title("raw")
+    # plt.show()
+
+    # # time series
+    # plt.plot(vols_chiral, label="with_chiral_restraint")
+    # plt.show()
+
+    # # compare filtered distributions
+    # plt.hist([x for x in vols_orig if x < 0], bins=np.linspace(-1, 1, 80), alpha=0.5, label="no_chiral_restr", density=True)
+    # plt.hist([x for x in vols_chiral if x < 0], bins=np.linspace(-1, 1, 80), alpha=0.5, label="with_chiral_restr", density=True)
+    # plt.legend()
+    # plt.title("pyramidal chirality")
+    # plt.xlabel("chiral volume")
+    # plt.ylabel("samples")
+    # plt.title("normalized")
+    # plt.show()
+
+
+class BaseTopologyRescaledCharges(topology.BaseTopology):
+    def __init__(self, scale, *args, **kwargs):
+        self.scale = scale
+        super().__init__(*args, **kwargs)
+
+    def parameterize_nonbonded(self, ff_q_params, ff_lj_params):
+        params, nb = topology.BaseTopology.parameterize_nonbonded(self, ff_q_params, ff_lj_params)
+        charge_indices = jnp.index_exp[:, 0]
+        new_params = jnp.asarray(params).at[charge_indices].multiply(self.scale)
+        return new_params, nb
+
+    def parameterize_nonbonded_pairlist(self, ff_q_params, ff_lj_params):
+        params, nb = topology.BaseTopology.parameterize_nonbonded_pairlist(self, ff_q_params, ff_lj_params)
+        charge_indices = jnp.index_exp[:, 0]
+        new_params = jnp.asarray(params).at[charge_indices].multiply(self.scale)
+        return new_params, nb
+
+
+def test_chiral_restraints_torsion():
```

Comment:
Add one-sentence description of what is being asserted
```suggestion
def test_chiral_restraints_torsion():
    """For a charge-scaled version of hydrogen peroxide, assert that:
    * without chiral bond restraints, cis/trans states are sampled ~ 30%/70%
    * with chiral bond restraints, ~ only specified state is sampled"""
```